src/backend/modules/transaction/transaction.service.ts

```diff
@@ -0,0 +1,282 @@
+import { Injectable } from "@nestjs/common";
+import { Cron, CronExpression } from '@nestjs/schedule';
+import { Prisma, PrismaClient } from "@prisma/client";
+import { PrismaService } from "../database/prisma.service";
+
+@Injectable()
+export class TransactionService {
+  constructor(private readonly prisma: PrismaService) { }
+
+  // @Cron(new Date(Date.now() + 5000))
+  @Cron(CronExpression.EVERY_5_MINUTES)
+  handleCron() {
+    // 0 - Active offer
+    // 1 - Expired offer
+    // 2 - User has no suffecient funds
+    // 3 - User has no stock
+    // 4 - Offer realized
+    // 5 - Transaction realized
+
+    this.checkOfferValidity()
+    this.transactionCycle()
+  }
+
+  async checkOfferValidity() {
+    await this.prisma.buyOffer.updateMany({
+      where: {
+        status: 0,
+        created: {
+          lte: new Date(Date.now() - 60 * 60 * 1000)
+        }
+      },
+      data: {
+        status: 1
+      }
+    })
+
+    await this.prisma.sellOffer.updateMany({
+      where: {
+        status: 0,
+        created: {
+          lte: new Date(Date.now() - 60 * 60 * 1000)
+        }
+      },
+      data: {
+        status: 1
+      }
+    })
+  }
+
+  async transactionCycle() {
+    const activeBuyOffers = await this.prisma.buyOffer.findMany({
+      where: {
+        status: 0
+      },
+      include: {
+        User: {
+          select: {
+            balanceCents: true
+          }
+        }
+      },
+      orderBy: {
+        created: 'asc'
+      }
+    })
+
+    for (const buyOffer of activeBuyOffers) {
+      await this.findMatchingSellOffers(buyOffer)
+    }
+  }
+
+  async findMatchingSellOffers(buyOffer: any) {
+    const { balanceCents } = buyOffer.User;
+    let { quantity } = buyOffer;
+
+    if (balanceCents < quantity * buyOffer.unitBuyPriceCents) {
+      await this.throwUserHasNoSufficientFundsError(buyOffer.buyOfferId)

```

Comment:
If you use "throw" word in function name, then maybe that function should throw error.